s3connector/s3.go

```diff
@@ -0,0 +1,133 @@
+package s3connector
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+type S3Config struct {
+	Endpoint    string `json:"endpoint"`
+	Region      string `json:"region"`
+	Bucket      string `json:"bucket"`
+	AccessKey   string `json:"accessKey"`
+	SecretKey   string `json:"secretKey"`
+	Prefix      string `json:"prefix"`
+	StorageType string `json:"storageType"`
+}
+
+type S3ObjectRange struct {
+	Start int64 `json:"start"`
+	End   int64 `json:"end"`
+}
+
+type S3ObjectPath struct {
+	Bucket string         `json:"bucket"`
+	Key    string         `json:"key"`
+	Range  *S3ObjectRange `json:"range,omitempty"`
+}
+
+type ObjectStorage interface {
+	StoreObject(key string, value io.ReadSeeker) (S3ObjectPath, error)
+	DeleteObject(key string) error
+	GetObject(objPath S3ObjectPath) (io.ReadCloser, error)
+	GetBucket() string
+}
+
+type s3ObjectStorage struct {
+	ObjectStorage
+	session      *session.Session
+	bucket       string
+	storageClass string
+	prefix       string
+}
+
+func NewS3ObjectStorage(config S3Config) (ObjectStorage, error) {
+	awsConf := &aws.Config{
+		Region:           aws.String(config.Region),
+		S3ForcePathStyle: aws.Bool(true),
```

Comment:
S3ForcePathStyle is needed only in case the "endpoint" is configured

Reply:
fixed